docs/localization.md

```diff
@@ -0,0 +1,138 @@
+[`graphs.h`]: https://github.com/microsoft/vcpkg-tool/blob/ca8099607bfa71adac301b56c601fd71d8ccab9b/include/vcpkg/base/graphs.h#L13
+[`graphs.cpp`]: https://github.com/microsoft/vcpkg-tool/blob/ca8099607bfa71adac301b56c601fd71d8ccab9b/src/vcpkg/base/graphs.cpp#L5
+
+### Message Names
+
+Message names should be descriptive and unique, and must be `CamelCase`.
+When referring to a message in `msg::format`, you must add `msg` to the front of it; for example,
+`DECLARE_MESSAGE(MyMessage, ...)` can be referred to as `msgMyMessage`.
+
+### Parameters
+
+If the message contains placeholders, each one must have a corresponding parameter.
+For example, `"Installing {package_name}:{triplet}"` should have the parameter list `(msg::package_name, msg::triplet)`.
+This allows us to make sure that we don't have format errors, where one forgets a placeholder,
+or adds an extra one.
+Each parameter in the list should be used in the message.
+
+### Comment
+
+The comment is used to give context to the translator as to what a placeholder will be replaced with.
+For example, in `BuildResultSummaryHeader` (`"SUMMARY FOR {triplet}"`),
+it will note that an example of `{triplet}` is `'x64-windows'`.
+Most message placeholders (see `messages.h` for the full list - search for `DECLARE_MSG_ARG`) have comments associated with them already.
+Only general placeholders like `{value}`, `{expected}`, and `{actual}` don't have comments associated with them.
+If you use any of these placeholders, write `example of {<name>} is '<expected-value>'` for each of those placeholders,
+separated by `\n`. You can also add context if you feel it's necessary as the first line.
+
+### Message
+
+Messages in vcpkg are written in American English. They should not contain:
+
+* formatting:
+  - indentation should be added with the `append_indent()` function;
+    if you need to add more than one indentation, you can use `append_indent(N)`
+  - newlines should be added with the `appendnl()` function
+  - Any other interesting characters (like `- ` for lists, for example) should use `append_raw(...)`
+* or for the prefixes:
+  - `"warning: "`, instead use `msg::format(msg::msgWarningMessage).append(msgMyWarning)`
+  - `"error: "`, instead use `msg::msgErrorMessage`
+  - `"internal error: "`, instead use `msg::msgInternalErrorMessage`.
+
+They should also not be simple, locale-invariant messages -- something like, for example,
+`{file}:{line}:{column}: ` should be done with `LocalizedString::from_raw(fmt::format("{}:{}:{}", file, line, column))`.
+
+Other than that, all messages that are printed to the user must be localizeable.
+However, that's as far as you need to go -- once you've done the writing of the message, and the `x-generate-default-message-map` stuff,
+you can rest easy not modifying anything else.
+
+### Putting it All Together
+
+Let's create a vcpkg hello world command to show off how one can use the messages API.
```

Comment:
I'm still not a fan of including a real example with a lot of code that's likely to rot rather than just linking to a real use but not over my dead body.